src/shared/conf-parser.c

```diff
@@ -221,8 +221,17 @@ static int parse_line(
                         return -ENOMEM;
 
                 if (sections && !nulstr_contains(sections, n)) {
+                        bool ignore = flags & CONFIG_PARSE_RELAXED;
 
-                        if (!(flags & CONFIG_PARSE_RELAXED) && !startswith(n, "X-"))
+                        ignore = ignore || startswith(n, "X-");
+
+                        if (!ignore) {
+                                _cleanup_free_ char *with_minus = NULL;
+                                with_minus = strjoin("-", n);
```

Comment:
oom check?

sounds easy to implement with with NULSTR_FOREACH without memory allocation